### ISSUE TITLE
Full offload over-reports VRAM by the logits buffer, which llama.cpp keeps in host memory

### ISSUE THREAD

**Issue by tobocop2** [gpustack/gguf-parser-go#40]
## Summary

The estimate reports more VRAM than a model uses when every layer is offloaded. On a model with a large vocabulary the error is about 700 MiB.

The cause is the logits buffer. llama.cpp keeps it in host memory. This estimator charges it to the graphics card, and charges it twice.

## Why this matters

A caller decides whether a model fits by reading this figure. An inflated figure rejects models that run. A 1.04 GiB model on a 2 GiB card reports that 512 tokens of context are left. The card really serves 13312 tokens.

## Where llama.cpp puts the buffer

`llama_context::output_reserve` picks the buffer type:

```c
auto * buft = ggml_backend_cpu_buffer_type();
// try to use the host buffer of the device where the output tensor is allocated
// for faster transfer to system memory
auto * output_dev_host_buft = output_dev ? ggml_backend_dev_host_buffer_type(output_dev) : nullptr;
if (output_dev_host_buft) { buft = output_dev_host_buft; }
buf_output.reset(ggml_backend_buft_alloc_buffer(buft, new_size));
```

It asks for a CPU buffer, or for the device's host buffer type. A host buffer type is pinned system memory. It is never device memory.

The engine prints the buffer at `-v`:

```
llama_context:  ROCm_Host  output buffer size =     0.58 MiB
```

The name ends in `_Host`, so the buffer is on the host. The size is `151936 x 4`, which is one row of logits. llama.cpp reserves the graph with `n_outputs = 1` for a normal decoder.

## Where this estimator puts it

`file_estimate__llamacpp.go:448`

```go
ob := a.EmbeddingLength * nOutputs * 4 /* float32 size */
if a.AttentionCausal {
    ob += a.VocabularyLength * nOutputs * 4 /* float32 size */
}
if fullOffload {
    e.Devices[idxOutputDevice].Footprint += GGUFBytesScalar(ob)
} else {
    e.Devices[0].Footprint += GGUFBytesScalar(ob)
}
```

`nOutputs` is `nTokens`, the whole batch. The estimate reserves 298.8 MiB where llama.cpp reserves 0.58 MiB, and it puts that reservation on the card.

`Computation.Output` at line 905 adds the identical quantity, and it also moves to the card at full offload. The logits are therefore counted twice.

## Measurements

Qwen3-0.6B Q8_0, 28 blocks. Offloading 29 layers puts everything on the card, which is the step that triggers the defect. The estimate is compared against the buffers llama.cpp reports for itself.

| card | the estimate adds | the card really gains |
|---|---|---|
| GTX 1070 Ti, CUDA | 738.0 MiB | 16.0 MiB |
| MI300X, ROCm | 738.5 MiB | 14.9 MiB |
| Intel UHD, Vulkan | 738.6 MiB | 16.9 MiB |

Three vendors agree. Qwen3-1.7B Q4_K_M on ROCm behaves the same way: the estimate adds 812.9 MiB and the card gains 28.5 MiB.

OLMoE-1B-7B, 16 blocks, full offload on ROCm: 4449.86 MiB reported against 4297.44 measured, an over-report of 152.42 MiB. The over-report appears only at full offload, because at partial offload the output device is already the host.

## The buffer is not allocated later

A 3000 token prompt grows device memory by 20 MiB on CUDA, and by 73 to 84 MiB on ROCm. Neither figure approaches 738 MiB. Every run answered a question.

## A second defect, filed separately

The partial offload figures carry a different defect. `--gpu-layers` selects different layers here than in llama.cpp. See the companion issue.

## How to reproduce

```
gguf-parser --path M --ctx-size 4096 --parallel 1 --gpu-layers N \
  --flash-attention --cache-type-k q8_0 --cache-type-v q8_0 \
  --platform-footprint 0,0 --json --skip-cache
```

Read `.estimate.items[0].vrams[0].nonuma`. Pass `--platform-footprint 0,0`, because the default adds an assumed 250 MiB driver reserve that llama.cpp does not allocate.

Measured values are the sum of the device model, KV and compute lines in the `-v` log. Take the last of each, because the log prints a probe pass first.

On AMD a raw `rocm-smi` reading is not comparable. It includes the HIP runtime, which measures 1039.9 MiB after load and is stable to 0.9 MiB across two models and two offload depths.

Metal uses unified memory, so the device figure and the host figure describe the same pool. The defect cannot show there.

**Comment by tobocop2** [comment]
Fixed on main. #43 folded in the commits from #41: the output buffer and the output projection stay on the host, eagle3 keeps its device placement, and both tests pin the behavior. A full offload no longer charges the logits buffer to the card.
